excellent/evaluator.go

```diff
@@ -238,7 +238,7 @@ func (s *xscanner) Scan() (xToken, string) {
 }
 
 // EvaluateExpression evalutes the passed in template, returning the raw value it evaluates to
-func EvaluateExpression(env utils.Environment, resolver utils.VariableResolver, template string) (interface{}, error) {
+func EvaluateExpression(env utils.Environment, resolver utils.Resolvable, template string) (interface{}, error) {
```

Comment:
Non-atom items in the context can now be `Resolvable` (i.e. can be keyed into), `Indexable` (i.e. can be indexed into) and/or `Atomizable` (i.e. can reduce themselves to an atom type)